melodies_monet/driver.py

```diff
@@ -362,6 +436,9 @@ def mask_and_scale(self):
                         self.obj[v].data = self.obj[v].where(self.obj[v] <= d['obs_max'])
                     if 'nan_value' in d:
                         self.obj[v].data = self.obj[v].where(self.obj[v] != d['nan_value'])
+
+                    # array needs to be writeable 
+                    self.obj[v].data = self.obj[v].data.copy()
                     
```

Comment:
I don't think we need this and if we did we should probably move it above the if statements. Can you remove your latest commit that added this in from this PR. Fine to use for your own analysis. If it comes up with other systems we can add this in at a later date.